### PR TITLE
Merge changes for 1.15.1 from the v1.15.x branch to dev

### PR DESCRIPTION
**Pull request by nmeyerhans**
### Summary

This PR ensures that changes that have already been staged for release as 1.15.1 are reflected in the dev branch for future feature releases.

In reality, this only pull in the addition of a unit test and a changelog update.

### Implementation details

   git checkout dev
   git merge v1.15.x

There were merge conflicts because there was a minimal change in 1.15.x to address #1067 while the same issue was fixed as part of a larger refactor in the dev branch. The conflict was resolved by keeping the fix from the dev branch.

### Testing
- [ ] Builds on Linux (`make release`)
- [ ] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [ ] Unit tests on Linux (`make test`) pass
- [ ] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [ ] Integration tests on Linux (`make run-integ-tests`) pass
- [ ] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [ ] Functional tests on Linux (`make run-functional-tests`) pass
- [ ] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Description for the changelog

CHANGELOG.md updates are included here as appropriate.

### Licensing

This contribution is under the terms of the Apache 2.0 License: yes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## Unreleased
 * Feature - Support pulling from Amazon ECR with specified IAM role in task definition
 * Feature - Enable support for task level CPU and memory constraints.
+* Bug - Fixed a bug where container state information wasn't reported. [#1067](https://github.com/aws/amazon-ecs-agent/pull/1067)
```
Comment (jhaynes) [inline CHANGELOG.md]: Should all three of these bug fixes be in the `## 1.15.1` header for this merge as well?

Comment (nmeyerhans) [reply CHANGELOG.md]: Yes, will update.

Comment (nmeyerhans) [reply CHANGELOG.md]: Dev already has the other fixes and their corresponding changelog. The merge just pulls in the one entry that was missing.

**Comment by nmeyerhans** [comment]
We may want to skip this merge and wait to merge all the 1.15.2 changes when they're ready. I don't see a need to merge this PR as is. Any disagreement?

**Comment by jhaynes** [comment]
Seems reasonable to me.

**Comment by richardpen** [comment]
yes, this was included in 1.15.2, we can just merge 1.15.2 to dev.

**Comment by nmeyerhans** [comment]
We merged all the 1.15.x changes up to and including 1.15.2 in #1082. We don't need this PR.

**Pull request closed without merging**